sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -40,6 +48,14 @@ public partial class ClientResult<T> : System.ClientModel.ClientResult
         public virtual T Value { get { throw null; } }
         public static implicit operator T (System.ClientModel.ClientResult<T> result) { throw null; }
     }
+    public abstract partial class EnumerableClientResult<T> : System.ClientModel.ClientResult, System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable, System.IDisposable where T : System.ClientModel.Primitives.IPersistableModel<T>
```

Comment:
The names don't seem symetrical. What about the following:

```csharp

AsyncEnumerableResult
EnumerableResult
//or
ClientResults
ClientResultsAsync
```

Reply:
I'm actually not wild about `ClientResults` because I think it complicates what we're defining a "result" to be (although I am open to debate here).

To take a concrete example, Pageable\<T\> is a collection of T's.  Does the "result" mean the T or with the response?  In a simple ClientResult\<T\> case there's a 1-1 relationship between them.  But ClientResult itself just exposes GetRawResponse so it seems like result maps more closely to the response than the T.  Given that, it feels strange to me to make the "result" plural when the T is the plural thing and there might only ever be one response.

Let me think.

Reply:
Looks like 3.5.2 in the FDG recommends using `Collection` in the name of types that derive from `IEnumerable<T>` ... what about `ClientResultCollection<T>` and `AsyncClientResultCollection<T>`?  Those are getting long, though, and don't really address my question about whether it is a collection of results, but I think I'm willing to drop argument since we defined ClientResult in the refdocs as "the result of a cloud service operation."  A sequence of such results could easily be a result collection.  Is it too long?